### ISSUE TITLE
Unable to compile 0.4.0 alpha on Linux (‘class QCustomPlot’ has no member named ‘addPlottable’)

### ISSUE THREAD

**Issue by IrreversibleReboot** [spillerrec/Overmix#147]
**Describe the bug**
I'm unable to compile Overmix 0.4.0 alpha on Linux by following the instructions provided by the readme. Here's the compile log:
```
Scanning dependencies of target Overmix
[ 68%] Building CXX object interface/CMakeFiles/Overmix.dir/Overmix_autogen/mocs_compilation.cpp.o
[ 70%] Building CXX object interface/CMakeFiles/Overmix.dir/cli/AlignParser.cpp.o
[ 71%] Building CXX object interface/CMakeFiles/Overmix.dir/cli/CommandParser.cpp.o
[ 72%] Building CXX object interface/CMakeFiles/Overmix.dir/cli/RenderParser.cpp.o
[ 73%] Building CXX object interface/CMakeFiles/Overmix.dir/cli/Processor.cpp.o
[ 74%] Building CXX object interface/CMakeFiles/Overmix.dir/gui/mainwindow.cpp.o
[ 75%] Building CXX object interface/CMakeFiles/Overmix.dir/gui/ImagesModel.cpp.o
[ 77%] Building CXX object interface/CMakeFiles/Overmix.dir/gui/FullscreenViewer.cpp.o
[ 78%] Building CXX object interface/CMakeFiles/Overmix.dir/gui/configs/AlignerConfigs.cpp.o
[ 79%] Building CXX object interface/CMakeFiles/Overmix.dir/gui/configs/RenderConfigs.cpp.o
[ 80%] Building CXX object interface/CMakeFiles/Overmix.dir/gui/savers/DumpSaver.cpp.o
[ 81%] Building CXX object interface/CMakeFiles/Overmix.dir/gui/visualisations/MovementGraph.cpp.o
/tmp/Overmix/interface/gui/visualisations/MovementGraph.cpp: In member function ‘void Line::addToPlot(QCustomPlot&, QColor)’:
/tmp/Overmix/interface/gui/visualisations/MovementGraph.cpp:54:9: error: ‘class QCustomPlot’ has no member named ‘addPlottable’; did you mean ‘hasPlottable’?
   54 |    plot.addPlottable( curve );
      |         ^~~~~~~~~~~~
      |         hasPlottable
make[2]: *** [interface/CMakeFiles/Overmix.dir/build.make:206: interface/CMakeFiles/Overmix.dir/gui/visualisations/MovementGraph.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:246: interface/CMakeFiles/Overmix.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
```

**How To Reproduce**
Just follow the instructions from the readme.

**Useful information:**
 - OS: Arch Linux
 - Overmix alpha 0.4.0
 - RAM: 15GB

Note: I tried to compile the latest git version too, but I kept getting errors about Waifu2x, and since it's mentioned in the release note that the later probably won't work, I gave up. I'm not all that familiar with C compiling, maybe there's a way to disable it that I've missed? Here's the log for completion's sake:
```
[ 34%] Building CXX object src/CMakeFiles/OvermixCore.dir/renders/DistanceMatrixRender.cpp.o
[ 35%] Building CXX object src/CMakeFiles/OvermixCore.dir/renders/PlaneRender.cpp.o
[ 36%] Building CXX object src/CMakeFiles/OvermixCore.dir/renders/StatisticsRender.cpp.o
[ 37%] Building CXX object src/CMakeFiles/OvermixCore.dir/renders/RobustSrRender.cpp.o
[ 37%] Building CXX object src/CMakeFiles/OvermixCore.dir/renders/EstimatorRender.cpp.o
[ 38%] Building CXX object src/CMakeFiles/OvermixCore.dir/renders/JpegRender.cpp.o
[ 39%] Building CXX object src/CMakeFiles/OvermixCore.dir/renders/JpegConstrainerRender.cpp.o
[ 40%] Building CXX object src/CMakeFiles/OvermixCore.dir/degraders/JpegDegrader.cpp.o
[ 40%] Building CXX object src/CMakeFiles/OvermixCore.dir/video/Video.cpp.o
[ 41%] Building CXX object src/CMakeFiles/OvermixCore.dir/video/VideoFrame.cpp.o
[ 42%] Building CXX object src/CMakeFiles/OvermixCore.dir/video/VideoStream.cpp.o
[ 43%] Building CXX object src/CMakeFiles/OvermixCore.dir/planes/manipulators/Waifu.cpp.o
In file included from /tmp/Overmix/src/planes/manipulators/Waifu.cpp:21:
/usr/include/w2xconv.h:31:10: fatal error: tchar.h: No such file or directory
   31 | #include "tchar.h"
      |          ^~~~~~~~~
compilation terminated.
make[2]: *** [src/CMakeFiles/OvermixCore.dir/build.make:791: src/CMakeFiles/OvermixCore.dir/planes/manipulators/Waifu.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:163: src/CMakeFiles/OvermixCore.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
```

**Comment by spillerrec** [comment]
The code used QCustomPlot version 1 in the past, but QCustomPlot v2 no longer needs that line and it should be removed. See commit 559697657ff1dd6cf0b6aa54eb8ea92b97b8eba1
0.4.0 alpha might have been using QCustomPlot version 1, I don't quite remember as it is 2 years old. Just removing the line might work.

But you should really use at least 0.4.0 alpha v2 or master since the other one is so old. There is a video importer now as well. The Waifu stuff shouldn't affect compiling, so I'm not sure what is going on there, but it should disable it automatically if `waifu2x-converter-cpp` is not installed on your system.
If you don't want to remove that package temporarily, you can edit [CMakeLists.txt](https://github.com/spillerrec/Overmix/blob/master/CMakeLists.txt) can remove the lines about `W2XC`. There is probably an way to make it an option you could toggle when calling `cmake`, but I don't know how to add that yet. (I'm currently working on my own AI models for upscaling, etc, so expect some AI stuff in the future.)

I would like to figure out why this is happening, but let me know if these workarounds work. I will take a closer look on a Linux box tomorrow.

**Comment by IrreversibleReboot** [comment]
Removing all the Waifu stuff from CMakeLists.txt allowed me to compile Overmix, thank you!
I do have waifu2x-converter-cpp installed on my system. I installed it using [this package](https://aur.archlinux.org/packages/waifu2x-converter-cpp/) from the AUR.
Let me know if there's anything I can do!

**Comment by IrreversibleReboot** [comment]
I'm also having a bit of trouble using Overmix, can I open a new issue for this?

**Comment by spillerrec** [comment]
Yes, feel free to do that

**Comment by spillerrec** [comment]
The compile issue is an error in waifu2x-converter-cpp and is fixed in the most recent 5.3.3 release which was released a month ago, but the AUR package has not been updated and is on 5.3.1. And they changed stuff again on master which broke it again, so I have pushed a fix so the git packages should work again.

I'm closing the issue now, just reopen it if you still experience issues.
